lib/vrl/compiler/src/expression/op.rs

```diff
@@ -43,8 +42,6 @@ impl Op {
                     rhs_span,
                     op_span,
                 });
-            } else if rhs_type_def.is_fallible() {
-                return Err(expression::Error::Fallible { span: rhs_span }.into());
```

Comment:
This case would cause issues when you chain error coalesced expressions (`a ?? b ?? c`) because this path would see `b` as being fallible, but wouldn't see that it would be made infallible through `?? c`.

Note that this check is purely cosmetic. I recently added this as an error message improvement, but doing `a ?? b` where `b` is fallible will still result in an error (reported because the root-level expression will be infallible), but it will be a tad less nice of an error.

I consider this okay for now, but it's something we can iterate on going forward.

Reply:
Does this still work?

```
., err |= something_fallible()
```

Reply:
It does, that's handled by the assignment expression.